### ISSUE TITLE
[Bug]: MTP does not mask embedding on position 0

### ISSUE THREAD

**Issue by DingYibin** [vllm-project/vllm#33299]
### 🐛 Describe the bug
<img width="2259" height="1039" alt="Image" src="https://github.com/user-attachments/assets/13b5d9c3-982c-461e-a8fc-e74de36d87c2" />
According to the report of deepseek v3, embedding on position 0 is dropped by left shifting instead of being set to 0.

```python
        # masking inputs at position 0, as not needed by MTP
        inputs_embeds = torch.where(positions.unsqueeze(-1) == 0, 0, inputs_embeds)
```
Codes like these do not follow the design of training.
### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by Yuxin1999** [comment]
Facing same problem, i have seen the input_ids already shifted in `set_inputs_first_pass`, therefore, the masked token will be t1 in MTP calculating, with input_ids: [0, t2, ..., tn+1], hidden_states[h0, ...., hn]

**Comment by Sunt-ing** [comment]
The shift that makes position 0 a valid input is in `SpecDecodeBaseProposer.set_inputs_first_pass` (`vllm/v1/spec_decode/llm_base_proposer.py`): `self.input_ids[: num_tokens - 1] = target_token_ids[1:]` with `positions` left unchanged, so the row at `positions == 0` already holds `emb(t1)`, a real token. Zeroing it diverges from the DeepSeek-V3 MTP design. vLLM's own `qwen3_next_mtp.py` and `qwen3_5_mtp.py` normalize `inputs_embeds` directly with no position-0 mask, so the unmasked form already ships in the tree.

I measured the impact with a substitute model (MiMo-7B-Base, the MTP model in `tests/v1/spec_decode/test_mtp.py`; not the DeepSeek model from the report).
Setup: vLLM main @1fd8bd02a, single RTX PRO 6000 (sm120, so `VLLM_USE_FLASHINFER_SAMPLER=0`), `enforce_eager=True`, `method=mtp`, greedy (`temperature=0`), `max_model_len=1024`.
Method: A/B by removing only the `inputs_embeds[positions == 0] = 0` line in `mimo_mtp.py` and adding it back, across `num_speculative_tokens` in {1, 2, 3} and 4 prompt sets (single-token, structured, natural-language, code), 2 repeats each, comparing the spec-decode acceptance counters from `LLM.get_metrics()`.
Result: per-config acceptance-rate deltas were mixed-sign and small (-2.1pp to +1.0pp), all within the run-to-run noise floor (identical-config reruns varied by up to 3.4pp, mean 0.8pp, from nondeterministic kernels). Pooled over all configs: 0.4029 accepted/drafted with the mask vs 0.4025 without it, a -0.03pp difference. Greedy output is unchanged because every draft token is verified by the target model.
So on this model the change is acceptance-neutral; it reads as a correctness/design-fidelity fix rather than a speedup.

@DingYibin #32592 already proposed the identical removal for deepseek/ernie/glm4_moe/glm4_moe_lite/mimo; reopening it (and extending to `glm_ocr_mtp.py`, `hy_v3_mtp.py`, and the `deepseek_v4` fused input RMSNorm, which postdate that PR) would close this.

**Comment by 2019598399** [comment]
I’ll take this. I’ll first audit the current MTP implementations to distinguish models using the shifted first-pass inputs from any with different position-zero semantics, then make the smallest correctness fix and add focused regression coverage. I’ll also account for the MTP implementations added after #32592.
